### PR TITLE
Propagate root subobjects setting to downsample indexes

### PR DESCRIPTION
**Pull request by kkrik-es**
Setting this in downsample tests allows running them with both trial and basic license.

### CODE REVIEW

File: x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java [hunk 1]
```diff
@@ -688,6 +689,12 @@ public static String createDownsampleIndexMapping(
             .utf8ToString();
     }
 
+    private static void addRootSubobjects(Map<String, Object> sourceIndexMappings, XContentBuilder builder) throws IOException {
+        if (sourceIndexMappings.containsKey("subobjects")) {
+            builder.field("subobjects", sourceIndexMappings.get("subobjects"));
+        }
```
Comment (kkrik-es) [inline x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java:695]: I'm on the fence for this one, it should be safe but we're updating production logic to make tests happy.. I can add tests if we like it, or guard behind a flag to use in tests only.

Comment (salvatore-campagna) [reply x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java:695]: This looks ok...anyway if you look a bit below we use a `MappingVisitor` object to visit mappings...maybe that is a better way to go through mappings and apply changes. Indeed I wonder if we need to do this for nested `subobjects` properties too.

Comment (kkrik-es) [reply x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java:695]: Right, though I was thinking of going to the other direction and skipping objects entirely. Today, we don't propagate any object properties so `subobjects:false` should always be applicable..

**Review by martijnvg — APPROVED** [review]
I left one comment, LGTM otherwise.

**Comment by kkrik-es** [comment]
Turns out the downsampling changes are not needed. I reverted them and both tests pass.

**Comment by kkrik-es** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/115577)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**